# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/WindowSurfaceWayland.cpp
# Commit: 09e90d1b7c7b
# Full Hash: 09e90d1b7c7b8342790e27a89c00b4aa32782e67
# Author: stransky <stransky@redhat.com>
# Date: 2021-06-17 04:27:31
# Regressor Bug: 1716850
# File Overlap Count: 1
# Description:
#   Bug 1716850 [Wayland] Don't lock WindowSurfaceWayland at WindowSurfaceWayland::Reset(), r=rmader
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D118078
# ==============================================================================

diff -r 399de9cecbef -r 09e90d1b7c7b widget/gtk/WindowSurfaceWayland.cpp
--- a/widget/gtk/WindowSurfaceWayland.cpp	Wed Jun 16 20:14:26 2021 +0000
+++ b/widget/gtk/WindowSurfaceWayland.cpp	Wed Jun 16 20:37:12 2021 +0000
@@ -815,11 +815,7 @@
 
 void WindowSurfaceWayland::Reset() {
   LOGWAYLAND(("WindowSurfaceWayland::Reset [%p]\n", this));
-  MutexAutoLock lock(mSurfaceLock);
-  if (mFrameCallback) {
-    wl_callback_destroy(mFrameCallback);
-    mFrameCallback = nullptr;
-  }
+  // No need to lock WindowSurfaceWayland here.
   mLastCommittedSurfaceID = -1;
 }
 
